src/main/java/com/aws/greengrass/util/GreengrassServiceClientFactory.java

```diff
@@ -70,6 +77,7 @@ private void validateConfiguration() {
             configValidationError = null;
         } catch (DeviceConfigurationException e) {
             configValidationError = e.getMessage();
+            logger.atWarn().log("Failed to validate config for Greengrass v2 data client: {}", configValidationError);
```

Comment:
Not sure I want the log here, I think putting it within the getter may be better, so that the warning and subsequent null pointer (or other error) are right next to each other

Reply:
updated